src/screen.c

```diff
@@ -2130,6 +2133,18 @@ if (db) fprintf(stderr, "initialize_raw_fb reset\n");
 	if (sscanf(str, "shm:%d", &shmid) == 1) {
 		/* shm:N */
 #if HAVE_XSHM || HAVE_SHMAT
+		/* try to use shm key*/
+		key_t shmkey = 0;
+		int newshmid = -1;
+		
+		shmkey = shmid;
+
+		newshmid = shmget(shmkey, 0, 0);
```

Comment:
I think this is OK now, my previous comment was also a bit erroneous: a new shared mem segment is only created if `IPC_CREAT` is specified in the flags.

Reply:
Yes. And IPC_EXCL is always used with IPC_CREAT. So setting shmflg as zero should be okay.
```
IPC_EXCL: This flag is used with IPC_CREAT to ensure that this call creates the segment.  If the segment already exists, the call fails.
```